lib/Alien/Build/Plugin/Core/Setup.pm

```diff
@@ -217,11 +218,25 @@ sub _cpu_arch {
   } elsif( $Config{archname} =~ m/
       \b darwin \b
     /ix ) {
-    my $cpu_brand = `sysctl -n machdep.cpu.brand_string`;
-    if( $cpu_brand =~ /Apple/ ) {
-      $arch = { name => 'aarch64' }; # Apple Silicon
-    } elsif( $cpu_brand =~ /Intel/ ) {
-      $arch = { name => 'x86_64' };  # Intel
+    my $hw_machine = `sysctl -n hw.machine`;
```

Comment:
```suggestion
    chomp( my $hw_machine = `sysctl -n hw.machine` );
```

Reply:
adding the chomp it works on G5 in 64 bit 

```
# $VAR1 = {
#           'cpu' => {
#                      'arch' => {
#                                  'hw.machine' => 'Power Macintosh',
#                                  'name' => 'ppc64'
#                                },
#                      'count' => 2
#                    },
#           'pkg-config' => {
#                             'pkgconf' => undef,
#                             'pkg-config' => '/opt/local/bin/pkg-config'
#                           },
#           'system_type' => 'unix',
#           'cmake_generator' => 'Unix Makefiles',
#           'compiler_type' => 'unix'
#         };
# pkg-config negotiate pick = PkgConfig::CommandLine
```

and 32 bit:

```
# ---
# cmake_generator: Unix Makefiles
# compiler_type: unix
# cpu:
#   arch:
#     hw.machine: Power Macintosh
#     name: ppc
#   count: 2
# pkg-config:
#   pkg-config: /opt/local/bin/pkg-config
#   pkgconf: ~
# system_type: unix
# pkg-config negotiate pick = PkgConfig::LibPkgConf
```

M1

```
# ---
# cmake_generator: Unix Makefiles
# compiler_type: unix
# cpu:
#   arch:
#     name: aarch64
#   count: 10
# pkg-config:
#   pkg-config: /opt/local/bin/pkg-config
#   pkgconf: /opt/local/bin/pkgconf
# system_type: unix
# pkg-config negotiate pick = PkgConfig::LibPkgConf
```

32 bit macOS 10.5 (I think I said 10.4 earlier, but it is actually 10.4)

```
# cmake_generator: Unix Makefiles
# compiler_type: unix
# cpu:
#   arch:
#     hw.machine: i386
#     name: x86
#   count: 2
# pkg-config:
#   pkg-config: /opt/local/bin/pkg-config
#   pkgconf: ~
# system_type: unix
# pkg-config negotiate pick = PkgConfig::CommandLine
# 
# 
# [config.site]
# # file automatically generated by /Users/ollisg/dev/Alien-Build/lib/Alien/Build/Plugin/Build/Autoconf.pm
# libdir='${prefix}/lib'
```

64bit intel:

```
# ---
# cmake_generator: Unix Makefiles
# compiler_type: unix
# cpu:
#   arch:
#     hw.machine: x86_64
#     name: x86_64
#   count: 8
# pkg-config:
#   pkg-config: /usr/local/bin/pkg-config
#   pkgconf: ~
# system_type: unix
# pkg-config negotiate pick = PkgConfig::LibPkgConf
```

Reply:
`name` looks right, but I am not sure why the `hw.machine` key sometimes shows up?

Reply:
I added that to debug the CI in a commit that I just removed.  Squashed it away now.